File: Algorithm/10_week_Algorithms/1.선형_알고리즘_기초/1B.js
// 문제1B-원소의 합 구하기


const readline = require('readline')

// 입력을 읽기 위한 인터페이스 설정
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})

// 입력을 저장할 변수
let input = []

// 각 줄의 입력을 처리
rl.on('line', (line) => {
  input.push(line)
})

// 모든 입력을 받은 후 처리
rl.on('close', () => {
  const n = parseInt(input[0].trim(), 10)
  const inputNumbers = input[1].trim().split(' ').map(Number)
  const answer = getSum(inputNumbers)
  console.log(answer)
})

// 합계를 계산하는 함수
function getSum(inputNumbers) {
  let sum = 0
  for (let num of inputNumbers) {
    sum += num
  }
  return sum
}

File: Algorithm/10_week_Algorithms/1.선형_알고리즘_기초/1B.c
// 문제1B-원소의 합 구하기


#include <stdio.h>

// 입력된 정수 배열의 합을 계산하는 함수
int getSum(int n, int inputNumbers[]) {
    int sum = 0;
    for (int i = 0; i < n; i++) {
        sum += inputNumbers[i];
    }
    return sum;
}

int main() {
    // 첫 번째 줄에서 데이터의 수n을 읽습니다.
    int n;
    scanf("%d", &n);

    int inputNumbers[n];

    // 두 번째 줄에서 공백으로 구분된 n개의 정수를 읽어 옵니다.
    for (int i = 0; i < n; i++) {
        scanf("%d", &inputNumbers[i]);
    }
    
    // 합계 계산
    int answer = getSum(n, inputNumbers);
    
    // 결과 출력
    printf("%d\n", answer);
    
    return 0;
}
